src/vivarium_gates_bep/components/metrics/mortality.py

```diff
@@ -0,0 +1,105 @@
+from collections import Counter
+
+import numpy as np
+
+from vivarium_public_health.metrics.mortality import MortalityObserver
+from vivarium_public_health.metrics.utilities import (QueryString, get_output_template, get_group_counts,
+                                                      get_deaths, get_years_of_life_lost)
+
+from vivarium_gates_bep.components.metrics.utilities import convert_whz_to_categorical
+
+
+class WHZMortalityObserver(MortalityObserver):
+
+    MortalityObserver.configuration_defaults.update(
+            {'metrics': {
+                'mortality_observer': {
+                    'by_whz': False
+            }}})
+
+
+    def __init__(self):
+        super().__init__()
+
+    @property
+    def name(self):
+        return 'whz_mortality_observer'
+
+    def setup(self, builder):
+        super().setup(builder)
+        # FIXME: does this need to be done with timedeltas?
+        self.step_size = builder.configuration.time.step_size / 365.25
+        # NOTE: Abie wants un-intervened on exposure to govern the groupings.
+        self.raw_whz_exposure = builder.value.get_value('child_wasting.exposure').source
+
+        if self.config.by_whz:
+            # We want to categorize based on WHZ at death, so we need to track that.
+            columns_required = ['alive', 'whz_at_death']
+            if self.config.by_age:
+                columns_required += ['age']
+            if self.config.by_sex:
+                columns_required += ['sex']
+            self.whz_at_death_view = builder.population.get_view(columns_required)
+
+            builder.population.initializes_simulants(self.on_initialize_simulants, requires_columns=columns_required)
+            builder.value.register_value_modifier('metrics', self.metrics)
+            builder.event.register_listener('time_step__prepare', self.on_time_step_prepare)
+            self.person_time = Counter()
+
+    def on_initialize_simulants(self, pop_data):
+        #pop = self.whz_at_death_view.subview(['alive']).get(pop_data.index)
+        pop = self.whz_at_death_view.get(pop_data.index)
```

Comment:
this perplexes me, I would think the commented line is what should work

Reply:
I thought part of the work James did was to eliminate the ambiguity that the subview method allowed for...